From: Elvis Angelaccio <elvis.angelaccio@kde.org>
Date: Fri, 30 Dec 2016 12:27:55 +0100
Subject: KMessageWidget: use darker red color when type is Error

Otherwise the close button is barely visible. There isn't an easy way to
change to color of the close button icon, so we use a slightly darker
background color instead.

BUG: 357210
FIXED-IN: 5.30

Differential Revision: D3418
---
 src/kmessagewidget.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/kmessagewidget.cpp b/src/kmessagewidget.cpp
index e8b25f6..3a6bfe0 100644
--- a/src/kmessagewidget.cpp
+++ b/src/kmessagewidget.cpp
@@ -284,6 +284,8 @@ void KMessageWidget::setMessageType(KMessageWidget::MessageType type)
         break;
     case Error:
         bg1.setRgb(218, 68, 83); // values taken from kcolorscheme.cpp (Negative)
+        // #357210: use darker color to improve the visibility of close button.
+        bg1 = bg1.darker(110);
         fg.setRgb(239, 240, 241);
         break;
     }
